docs/release-notes/self-hosted/self-hosted-v12.0.0.md

```diff
@@ -0,0 +1,128 @@
+
+## Upgrading Codacy Self-hosted
+
+Follow the steps below to upgrade to Codacy Self-hosted v12.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v12.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:
+
+    -   [Codacy Analysis CLI 7.8.3](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.8.3)
+    -   [Codacy Coverage Reporter 13.13.1](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.13.1)
+
+## Breaking changes
+
+-   If you have set `scheduler.enable = "true"` in `values.yaml`, do the following before updating Kubernetes.
+
+    Assuming you're using the default `codacy` namespace:
+
+    1.  Remove all `scheduler.*` settings from `values.yaml`
+
+    1.  Update the ConfigMap of `worker-manager`:
+
+        ```bash
+        kubectl patch configmap/worker-manager --namespace codacy --type merge --patch '{"data":{"CONFIG_FORCE_codacy_kubernetes_scheduler_enable": "false","CONFIG_FORCE_codacy_kubernetes_scheduler_name":"default-scheduler"}}'
+        ```
+
+    1.  Restart the `worker-manager` Deployment:
+
+        ```bash
+        kubectl rollout restart deployment/worker-manager -n codacy
+        ```
+
+    1.  List any pending `worker` Pods:
+
+        ```bash
+        kubectl get pods -l app=worker --field-selector=status.phase=Pending -n codacy
+        ```
+
+    1.  If there are any pending `worker` Pods, remove them:
+
+        !!! warning
+            This is a destructive action. Make sure you understand the consequences.
+
+        ```bash
+        kubectl delete pods -l app=worker --field-selector=status.phase=Pending -n codacy
+        ```
+
+-   If you're upgrading your Kubernetes installation to v1.25, make sure to do the following in the recommended order to avoid service outages:
+
+    1.  Upgrade to Codacy Self-hosted v12.0.0
+
+    1.  If using the monitoring functionality [Grafana + Prometheus + Loki](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#setting-up-monitoring-using-grafana-prometheus-and-loki), update Loki and Promtail by following the latest respective installation instructions ([Loki](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#2-installing-loki), [Promtail](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#3-installing-promtail)).
+
+    1.  Upgrade Kubernetes to v1.25
+
+## Product enhancements
+
+-   Added support for [Kubernetes 1.25.\*](https://docs.codacy.com/v12.0/chart/requirements/#kubernetes-or-microk8s-cluster-setup) (REL-1306)
```

Comment:
Should we add another line mentioning adding support for kubernetes 1.24 or we'll consider it implied since we have support for 1.25? Checked the release notes for sh 11 and we only mentioned support for 1.23 in those.

Reply:
Fine, we'll proceed with option 2.

Reply:
@tfrgoncalves We already added support to K8s 1.24 on [SH v12 release notes (current PR)](https://feat-release-notes-self-hosted-12-0--docs-codacy.netlify.app/release-notes/self-hosted/self-hosted-v12.0.0/#product-enhancements), and removed K8s 1.24 from [SH v11 requirements](https://docs.codacy.com/v11.0/chart/requirements/#kubernetes-or-microk8s-cluster-setup).

Can you please validate and give us approval?